services/web_crawling.py

```
@@ -0,0 +1,61 @@
+
+    def __init__(self):
+        self.base_url = config.WEB_CRAWLER_HTTP 
+
+    def get_web_content(self, url, username=None, password=None):
+        if not url.startswith("https://"):
+            url = "https://" + url
+        credentials = f"&username={username}&password={password}" if username and password else ""
+        request_url = f"{self.base_url}parse?url={url}{credentials}"
+        
+        try:
+            response = requests.get(request_url)
```

Ruff (0.13.3)

Probable use of `requests` call without timeout

(S113)